projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -218,8 +211,17 @@ public function get_item_schema() {
 	private function prepare_menu_item( array $menu_item ) {
 		global $submenu;
 
-		// Exclude unauthorized menu items.
-		if ( ! current_user_can( $menu_item[1] ) ) {
+		$current_user_can_access_menu = current_user_can( $menu_item[1] );
+		$has_first_menu_item          = isset( $submenu[ $menu_item[2] ][0] );
+		$menu_slug                    = $menu_item[2];
+
+		// Exclude unauthorized menu items when the user does not have access to the menu and the first submenu item.
+		if ( ! $current_user_can_access_menu && $has_first_menu_item && ! current_user_can( $submenu[ $menu_slug ][0][1] ) ) {
+			return array();
+		}
+
+		// Exclude unauthorized menu items that don't have submenus.
```

Comment:
Is this comment accurate? I think it's technically possible to have a submenu with the `0` position unset with other positions like `1` set, and in such cases we won't be excluding the menu item.

Reply:
Oh, yeah, you are right, thanks! :)